### ISSUE TITLE
New Resource for VPC

### ISSUE THREAD

**Issue by orlandothoeny** [cloudamqp/terraform-provider-cloudamqp#133]
To be able to manage the whole lifecycle of a VPC, a new resource type would be good.

There is currently no way to delete a VPC using this provider. It either has to be done using the API or the web UI.

**Comment by AlexRex** [comment]
Also for me, it looks like after destroying the stack the VPC stills there. I guess is because Terraform doesn't know about this VPC at all..

**Comment by luiselizondo** [comment]
Actually I'm seeing something different, the VPC is destroyed after the instance is deleted and if the VPC is left with no instances at all. Managing the creation of VPCs should be a must for this plugin.

**Comment by dentarg** [comment]
Yes, until we have added the resource to manage VPCs, empty VPCs are automatically deleted when you remove an instance using the API. This behaviour may change in the future.

**Comment by tbroden84** [comment]
We have done development and testing to add the managed VPC resource. Unfortunately will introduce breaking changes and affect multiple resources. After internal discussion we decided move forward to create a new major version for this.

Example of breaking changes:
- Instance: will never remove VPC when deleted. (Previous behaviour due to our underlying API). 
- Instance: no longer able to create VPC
- VPC peering, info: dependency to VPC resource (Previous dependency to instance).

**Comment by tbroden84** [comment]
After even more internal discussion and testing. We found out it was possible to make it work without breaking changes. We will release v1.16.0 with the new managed VPC resource. Then put the above mention breaking changes into the next major release.

**Comment by tbroden84** [comment]
Just released [v1.16.0](https://github.com/cloudamqp/terraform-provider-cloudamqp/releases/tag/v1.16.0) with managed VPC resource.
